src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.h

```diff
@@ -24,36 +24,79 @@ typedef enum EnDinofosBodyPart {
     /* 12 */ DINOFOS_BODYPART_MAX
 } EnDinofosBodyPart;
 
+typedef enum EnDinofosJumpType {
+    /* 0 */ DINOFOS_JUMP_TYPE_FORWARD,
+    /* 1 */ DINOFOS_JUMP_TYPE_IN_PLACE,
+    /* 2 */ DINOFOS_JUMP_TYPE_BACKWARD,
+    /* 3 */ DINOFOS_JUMP_TYPE_SLASH
+} EnDinofosJumpType;
+
+/**
+ * The elements before DINOFOS_COLLIDER_FIRE_START_INDEX are for a Dinofos' body in the actor's collider.
+ * The last three elements are for the flame.
+ */
+#define DINOFOS_COLLIDER_FIRE_START_INDEX 6
+
+/**
+ * unk_294 explanation :
+ *
+ * The best way to compare is to look from timer1 to circlingRate in this current actor struct and from
+ * 0x03F0 to 0x0408 in OoT's struct for Lizalfos/Dinolfos.
+ *
+ * In OoT, an additionnal ice timer, booleans related to the platforms fight and platforms indices are stored
+ * after the two timers EnDinofos still has (all integers). Moreover, the alpha value (envColorAlpha) has been moved up
+ * in MM.
+ *
+ * In MM, subCamId and subCamRot were added to handle the new cutscenes. Also, the effectIndex for blur is new.
+ *
+ * All in all, unk_294 is probably remnants of the now unnecessary variables mentioned above.
+ */
+
 typedef struct EnDinofos {
     /* 0x000 */ Actor actor;
     /* 0x144 */ SkelAnime skelAnime;
     /* 0x188 */ Vec3s jointTable[DINOLFOS_LIMB_MAX];
     /* 0x206 */ Vec3s morphTable[DINOLFOS_LIMB_MAX];
     /* 0x284 */ EnDinofosActionFunc actionFunc;
-    /* 0x288 */ u8 unk_288;
-    /* 0x289 */ u8 unk_289;
+    /* 0x288 */ u8 envColorAlpha;
+    /* 0x289 */ u8 eyeTexIndex;
     /* 0x28A */ u8 drawDmgEffType;
-    /* 0x28B */ u8 unk_28B;
-    /* 0x28C */ s16 unk_28C;
-    /* 0x28E */ s16 unk_28E;
-    /* 0x290 */ s16 unk_290;
-    /* 0x292 */ s16 unk_292;
+    /* 0x28B */ u8 isDodgingGoronPound;
+    /* 0x28C */ s16 targetRotY;
+    /* 0x28E */ s16 headRotY;
+    /* 0x290 */ union { // multi-use timer
+        s16 timer1;
+        s16 cutsceneTimer;
+        s16 attackTimer;
+        s16 actionTimer;
+        s16 headTimer;
+        s16 stunTimer;
+        s16 jumpType;
+        s16 isJumpingBackward;
+    };
+    /* 0x292 */ union { // second multi-use timer
+        s16 timer2;
+        s16 sidestepTimer;
+        s16 walkTimer;
+        s16 idleTimer;
```

Comment:
This is not something I'm asking or demanding you to do; this is just something I was thinking about. But one thing I've noticed is that the developers sometimes like to give certain kinds of movement their own timer. I first noticed it with Garo Master; you can see a variable called `attackMovementTimer` because it's used for dash attacks and teleport attacks. Looking at this actor, I can sort of see something similar here. These timers all seem to be related to movement of some sort. Even `idleTimer` is sort of movement related in the sense that it's measuring how long it needs to wait before it makes a move.

The other timer doesn't really seem to be tied to movement, at least directly. For example, `attackTimer` is used for the jump attack, but that's a jumping *attack*; there's more to it than just the movement. So maybe we could collapse all of these down into a single `movementTimer`? You would still likely want the union for `timer2`, which as I stated before, I have *no* clue how to name that one. But perhaps that would simplify things by making the functions a little more consistent. It might *complicate* things, though, by using "movement" timer for idling, which is sort of the opposite of moving. 

What do you think? Again, if you prefer it as-is, that's fine too. I just saw this as potentially similar to something I had seen before.